### PR TITLE
Fix nested inputs aliasing

### LINKED ISSUE #331: Nested filter on related model not working
(Prisma2/nexus)

I am using the following (simplified) model:
```graphql
model Staff {
  id         String       @default(cuid()) @id @unique
  user       User?
  active     Boolean?
}
model User {
  id         String        @default(cuid()) @id @unique
  email      String        @unique
  staff      Staff?
}
```
Using photon, the following query works:
```ts
  const staff = await photon.users.findMany({
    where: {
      staff: { active: true },
    },
  });
```
However, the following graphql query, exposed through nexus-prisma, fails:
```graphql
query {
  findManyUser(
    where: { staff: { active: true } }
  ) {
    id
    email
  }
}
```
with the error message:
```
Field "active" is not defined by type QueryFindManyUserWhereInput.
```
The nexus-prisma type generation seems to be incorrect, `active` should not be of type `QueryFindManyUserWhereInput`, I think.

### PR DESCRIPTION
**Pull request by Weakky**
Fixes #331

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -27,7 +42,7 @@ export function flatMap<T, U>(
 }
 
 export function nexusOpts(param: {
-  type: string
+  type: core.AllInputTypes | core.AllNexusInputTypeDefs<string>
```
Comment (jasonkuhrt) [inline src/utils.ts]: What's the difference between these two types?

File: tests/schema.test.ts [hunk 2]
```diff
@@ -289,3 +289,46 @@ test("it does not expose pagination", async () => {
 
   expect(printSchema(schema)).toMatchSnapshot()
 })
+
+test('alias top-level input types only if not customized', async () => {
```
Comment (jasonkuhrt) [inline tests/schema.test.ts]:
```suggestion
test('renames top-level input types that are customized', async () => {
```

Or do I misunderstand?

File: tests/__snapshots__/schema.test.ts.snap [hunk 3]
```diff
@@ -1,5 +1,78 @@
...
+input PostWhereInput {
+  id: StringFilter
+  AND: [PostWhereInput!]
+  OR: [PostWhereInput!]
+  NOT: [PostWhereInput!]
+  author: UserWhereInput
+}
+
+type Query {
+  posts(where: QueryFindManyPostWhereInput, skip: Int, after: String, before: String, first: Int, last: Int): [Post!]
```
Comment (jasonkuhrt) [inline tests/__snapshots__/schema.test.ts.snap]: Aside: strikes me that this should be named `QueryPostsWhereInput`

File: src/nexus-prisma/NexusPrismaBuilder.ts [hunk 4]
```diff
@@ -262,59 +266,69 @@ export class NexusPrismaBuilder {
...
+    const type = this.dmmf.getInputType(inputTypeName)
+
+    // Do not alias type name if type is not customized
+    if (input === true) {
+      return type
+    }
+
+    const argsNames = Object.keys(input)
+
+    // Alias type name if some fields are whitelisted
```
Comment (jasonkuhrt) [inline src/nexus-prisma/NexusPrismaBuilder.ts]:
```suggestion
    // Rename the InputObject type definition if some fields are whitelisted
```

Feel like `alias` is not the word we want 🤔

File: src/nexus-prisma/NexusPrismaBuilder.ts [hunk 5]
```diff
@@ -229,31 +235,29 @@ export class NexusPrismaBuilder {
...
-            whereArg.inputType.type,
-            whereArg.inputType.kind,
-          )
-        ] = Object.keys(opts.filtering).filter(
-          fieldName => (opts.filtering as any)[fieldName] === true,
+          `Could not find filtering argument in ${prismaModelName}.${
+            field.name
+          }(${field.args
+            .map(a => `${a.name}: ${a.inputType.type}`)
+            .join(', ')}): ${field.outputType.type}`,
```
Comment (jasonkuhrt) [inline src/nexus-prisma/NexusPrismaBuilder.ts]: Bit hard to read this

**Review by jasonkuhrt — APPROVED** [review]

File: src/nexus-prisma/NexusPrismaBuilder.ts (hunk 5 above)
Comment (Weakky) [reply src/nexus-prisma/NexusPrismaBuilder.ts]: Agreed but not sure how to factor it better. It generates something like `Query.posts(first: Int, last: Int, before: String, after: String)`

Comment (Weakky) [reply src/nexus-prisma/NexusPrismaBuilder.ts]: I reverted it anyway, we'll think about better errors in another pr

**Pull request merged**